themes/finna2/js/finna-record.js

```diff
@@ -163,11 +163,41 @@ finna.record = (function finnaRecord() {
     });
   }
 
+  function wayfinderPlacementLinkLookup(element) {
+    const url = VuFind.path + '/AJAX/JSON?' + new URLSearchParams({
+      method: 'wayfinderPlacementLinkLookup',
+      placement: element.dataset.location
+    });
+
+    fetch(url)
+      .then(response => response.json())
+      .then(responseJSON => {
+        let wayfinder_link = document.createElement('a');
+        wayfinder_link.className = 'icon-link__label';
+        wayfinder_link.setAttribute('target', '_blank');
+        wayfinder_link.setAttribute('href', responseJSON.data.marker_url);
+        wayfinder_link.innerHTML = VuFind.icon('map-marker');
+
+        let reader_tag = document.createElement('span');
+        reader_tag.className = 'sr-only';
+        reader_tag.innerHTML = 'View item placement on interactive map';
```

Comment:
This needs translation. If you don't want to add a new translation, you can use the existing `location-service` translation like this: `reader_tag.innerHTML = VuFind.translate('location-service');` Otherwise you'll need to add the translation to local/language/finna/fi.ini etc. and to themes/finna2/templates/layout/finna-js-translations.phtml to make it available in JS. Though frankly, it would be much easier to add the whole link as a template in the HTML and just fill in the URL etc. when lookup is done.

Reply:
Since you already render the element with the js-wayfinder-placeholder class in [holding-callnumber.phtml](https://github.com/NatLibFi/NDL-VuFind2/pull/2684/files/bc5406567fecda61cf33f7b4e2ba2b512d1fad81#diff-1f275614a6e8ef4bae9ccec547201c447654e47e84278aa925b3da73ded1755eR27), I'd include everything there. Something like this (N.B. completely untested):

```
    <div class="js-wayfinder-placeholder" data-location="<?=$this->htmlSafeJsonEncode($this->wayfinder)?>">
      <template class="js-wayfinder-link">
        <a class="icon-link__label" target="_blank" href="#">
          <?=$this->icon('map-marker')<span class="sr-only"><?=$this->transEsc('Location Service')?></span>
        </a>
      </template>
    </div>
```
Then just take the template here and replace href:
```
        let link_template = element.querySelector('.js-wayfinder-link');
        if (!link_template) {
          element.remove();
          return;
        }
        let link_container = link_template.cloneNode(true);
        let link = link_container.content.querySelector('a');
        if (!link) {
          element.remove();
          return;
        }
        link.setAttribute('href', responseJSON.data.marker_url);
        element.innerHTML = link_container.innerHTML;
```

Reply:
Returning an HTML snippet is also frequently used, but since the content is quite simple and doesn't require much logic, I think this is the simplest approach.